lib/puppet/functions/ldapquery/query.rb

```diff
@@ -11,9 +11,16 @@
   local_types do
     type <<~TYPE_ALIAS
       Options = Struct[
-        Optional[base]   => String[1],
-        Optional[scope]  => Enum['sub','base','single'],
-        Optional[server] => String[1],
+        Optional[base]           => String[1],
```

Comment:
Have been looking at this and also #47  which differ in that `base` is it's own parameter.

This looks good other than of course docs and spec but 

There is a case to make scope and base their own parameters so the options becomes only the connection parameters.